coda_v2_python_client/firebase_client_wrapper.py

```diff
@@ -367,3 +368,71 @@ def set_segment_messages_metrics(self, segment_id, messages_metrics):
         :type metrics_map: core_data_modules.data_models.metrics.MessagesMetrics
         """
         self.get_segment_messages_metrics_ref(segment_id).set(messages_metrics.to_firebase_map())
+
+    def compute_segment_coding_progress(self, segment_id, messages=None):
+        """
+        Compute and return the progress metrics for a given dataset.
+
+        This method will initialise the counts in Firestore if they do not already exist.
+
+        :param segment_id: Id of a segment.
+        :type segment_id: str
+        :param messages: list of core_data_modules.data_models.message.Message, defaults to None
+        :param messages: If specified, it computes progress metrics based on the provided messages
+                         else it downloads messages from the requested segment. Defaults to None.
+        :type messages: core_data_modules.data_models.message.Message | None
+        :return: Messages metrics.
+        :rtype: core_data_modules.data_models.metrics.MessagesMetrics
+        """
+        if messages is None:
+            messages = self.get_segment_messages(segment_id)
+        messages_with_labels = 0
+        wrong_scheme_messages = 0
+        not_coded_messages = 0
+
+        code_schemes = {code_scheme.scheme_id: code_scheme for code_scheme in self.get_all_code_schemes(segment_id)}
+
+        for message in messages:
+            # Get the latest label from each scheme
+            latest_labels = {label.scheme_id: label for label in message.get_latest_labels()} # of scheme id -> label
```

Comment:
I don't understand why this needs to be a look-up table, as you only use it with latest_labels.values()
Can't this just be
```suggestion
            latest_labels = message.get_latest_labels()
```
?